pwndbg/elf.py

```diff
@@ -296,16 +296,10 @@ def getjmpslots(path):
     program = pwndbg.which.which("readelf")
     if program:
         argv = [program, "-r", path]
-        readelf_out = subprocess.check_output(argv).decode('utf-8')
-        if "Error" in readelf_out:
-            return "Error"
-        else:
-            jmpslots = ""
-            for line in readelf_out.splitlines():
-                if "JUMP" not in line:
-                    continue
-                else:
-                    jmpslots = jmpslots + line + "\n"
-            return jmpslots
+        try:
+            readelf_out = subprocess.check_output(argv).decode('utf-8')
+        except: raise OSError("Error during readelf execution.")
+        relocations = filter(lambda l: "JUMP" in l, readelf_out.splitlines())
```

Comment:
```
[dc@dc:~]$ readelf -r /bin/ls

Relocation section '.rela.dyn' at offset 0x16e0 contains 8 entries:
  Offset          Info           Type           Sym. Value    Sym. Name + Addend
00000061eff0  003300000006 R_X86_64_GLOB_DAT 0000000000000000 __libc_start_main@GLIBC_2.2.5 + 0
00000061eff8  003e00000006 R_X86_64_GLOB_DAT 0000000000000000 __gmon_start__ + 0
00000061f5e0  006c00000005 R_X86_64_COPY     000000000061f5e0 __progname@GLIBC_2.2.5 + 0
00000061f5e8  007800000005 R_X86_64_COPY     000000000061f5e8 stdout@GLIBC_2.2.5 + 0
00000061f5f0  007a00000005 R_X86_64_COPY     000000000061f5f0 optind@GLIBC_2.2.5 + 0
00000061f5f8  007500000005 R_X86_64_COPY     000000000061f5f8 optarg@GLIBC_2.2.5 + 0
00000061f600  006f00000005 R_X86_64_COPY     000000000061f600 __progname_full@GLIBC_2.2.5 + 0
00000061f620  007200000005 R_X86_64_COPY     000000000061f620 stderr@GLIBC_2.2.5 + 0

Relocation section '.rela.plt' at offset 0x17a0 contains 108 entries:
  Offset          Info           Type           Sym. Value    Sym. Name + Addend
00000061f018  000100000007 R_X86_64_JUMP_SLO 0000000000000000 __ctype_toupper_loc@GLIBC_2.3 + 0
00000061f020  000200000007 R_X86_64_JUMP_SLO 0000000000000000 __uflow@GLIBC_2.2.5 + 0
00000061f028  000300000007 R_X86_64_JUMP_SLO 0000000000000000 getenv@GLIBC_2.2.5 + 0
00000061f030  000400000007 R_X86_64_JUMP_SLO 0000000000000000 cap_to_text + 0
00000061f038  000500000007 R_X86_64_JUMP_SLO 0000000000000000 sigprocmask@GLIBC_2.2.5 + 0
00000061f040  000600000007 R_X86_64_JUMP_SLO 0000000000000000 __snprintf_chk@GLIBC_2.3.4 + 0
00000061f048  000700000007 R_X86_64_JUMP_SLO 0000000000000000 raise@GLIBC_2.2.5 + 0
00000061f050  006e00000007 R_X86_64_JUMP_SLO 0000000000402260 free@GLIBC_2.2.5 + 0
00000061f058  000800000007 R_X86_64_JUMP_SLO 0000000000000000 abort@GLIBC_2.2.5 + 0
00000061f060  000900000007 R_X86_64_JUMP_SLO 0000000000000000 __errno_location@GLIBC_2.2.5 + 0
00000061f068  000a00000007 R_X86_64_JUMP_SLO 0000000000000000 strncmp@GLIBC_2.2.5 + 0
00000061f070  008100000007 R_X86_64_JUMP_SLO 00000000004022a0 localtime_r@GLIBC_2.2.5 + 0
00000061f078  000b00000007 R_X86_64_JUMP_SLO 0000000000000000 _exit@GLIBC_2.2.5 + 0
00000061f080  000c00000007 R_X86_64_JUMP_SLO 0000000000000000 strcpy@GLIBC_2.2.5 + 0
(...)
```
This should check for `JUMP` in `Type` column instead of all line.